helm-chart-sources/pulsar/templates/broker-sts/broker-sts-configmap.yaml

```diff
@@ -81,17 +81,52 @@ data:
+  {{- if .Values.openid.enabled }}
+  brokerClientAuthenticationPlugin: "org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2"
+  brokerClientAuthenticationParameters: "{\"privateKey\":\"file:///pulsar/openid-client-credentials/superuserKey.json\",\"issuerUrl\":\"{{ .Values.openid.issuerUrl }}\",\"scope\":\"{{ .Values.openid.scope }}\"}"
+  {{- else }} # There's probably a smarter way to write this so it's more readable but gets trimmed to one line so Jackson doesn't blow up on new lines when parsing.
   brokerClientAuthenticationPlugin: "org.apache.pulsar.client.impl.auth.AuthenticationToken"
   brokerClientAuthenticationParameters: "file:///pulsar/token-superuser/superuser.jwt"
-  {{- if .Values.brokerSts.extraAuthProvider }}
+  {{- end }}
+  {{- if .Values.brokerSts.extraAuthProvider }} # There surely is a cleaner way to do all this...
+  {{- if .Values.openid.enabled }}
+  authenticationProviders: "{{ .Values.brokerSts.authenticationProviders }},{{ .Values.brokerSts.extraAuthProvider }},{{ .Values.openid.authenticationProvider }}"
+  {{- else }}
   authenticationProviders: "{{ .Values.brokerSts.authenticationProviders }},{{ .Values.brokerSts.extraAuthProvider }}"
+  {{- end }}
+  {{- else }}
+  {{- if .Values.openid.enabled }}
+  authenticationProviders: "{{ .Values.brokerSts.authenticationProviders }},{{ .Values.openid.authenticationProvider }}"
   {{- else }}
   authenticationProviders: "{{ .Values.brokerSts.authenticationProviders }}"
   {{- end }}
+  {{- end }}
   {{- if .Values.keycloak.enabled }}
   PULSAR_PREFIX_openIDAllowedTokenIssuers: "{{ template "pulsar.get.http.or.https" . }}{{ template "pulsar.keycloak.fullname" .}}{{ template "pulsar.keycloak.issuer.port" .}}/realms/{{ .Values.keycloak.realm }},{{ template "pulsar.get.http.or.https" . }}{{ template "pulsar.keycloak.fullname" .}}.{{ .Release.Namespace }}{{ template "pulsar.keycloak.issuer.port" .}}/realms/{{ .Values.keycloak.realm }},{{ template "pulsar.get.http.or.https" . }}{{ template "pulsar.keycloak.fullname" .}}.{{ template "pulsar.serviceDnsSuffix" . }}{{ template "pulsar.keycloak.issuer.port" .}}/realms/{{ .Values.keycloak.realm }}"
   {{- end }}
   {{- end }}
+
+{{- if .Values.openid.enabled }}
+  openIDAllowedTokenIssuers: "{{ .Values.openid.issuerUrl }}" # Does this need PULSAR_PREFIX_ ? 
+  PULSAR_PREFIX_openIDAllowedTokenIssuers: "{{ .Values.openid.issuerUrl }}"
+  openIDAllowedAudience: "{{ .Values.openid.allowedAudience }}"
+  PULSAR_PREFIX_openIDAllowedAudience: "{{ .Values.openid.allowedAudience }}"
+  openIDRoleClaim: "{{ .Values.openid.roleClaim }}"
+  openIDAcceptedTimeLeewaySeconds: "0"
+  openIDJwkCacheSize: "10"
+  openIDJwkExpiresSeconds: "300"
+  openIDJwkConnectionTimeoutMillis: "10000"
+  openIDJwkReadTimeoutMillis: "10000"
+  openIDMetadataCacheSize: "10"
+  openIDMetadataExpiresSeconds: "86400"
+  openIDMetadataConnectionTimeoutMillis: "10000"
+  openIDMetadataReadTimeoutMillis: "10000"
+  openIDRequireHttps: "true"
```

Comment:
We could probably set these once in the values file and then let users override them. I'd expect them to be the same for every component, so no need to make the config per component.

Also, all of these need to be prefixed with `PULSAR_PREFIX_`.

Reply:
@michaeljmarshall PTAL.